### PR TITLE
🖍  Hide the system layer and progress bar in preview mode

### PR DESCRIPTION
**Pull request by coreymasanto**
- Part of #37129

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story.js [hunk 1]
```diff
@@ -1699,6 +1699,13 @@ export class AmpStory extends AMP.BaseElement {
     const vState = this.getAmpDoc().getVisibilityState();
     const isPreview = vState === VisibilityState_Enum.PREVIEW;
     const isVisible = vState === VisibilityState_Enum.VISIBLE;
+
+    if (isPreview) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.js:1703]: Will this run on page load, or only when the state changes?

File: extensions/amp-story/1.0/amp-story.js [hunk 2]
```diff
@@ -1699,6 +1699,13 @@ export class AmpStory extends AMP.BaseElement {
...
+      this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, false);
+    } else if (isVisible) {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.js:1705]: Sorry about this, just a personal and unjustified preference (I usually struggle to follow `else` statements)

```suggestion
    }
    if (isVisible) {
```

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 3]
```diff
@@ -92,105 +92,115 @@ const HIDE_MESSAGE_TIMEOUT_MS = 1500;
...
+ * @param {boolean} isHidden
  * @return {!Element}
  */
-const renderSystemLayerElement = (element, children) => (
-  <aside class="i-amphtml-story-system-layer i-amphtml-story-system-reset">
-    {children}
-    <a class={String(ATTRIBUTION_CLASS)} target="_blank">
-      <div class="i-amphtml-story-attribution-logo-container">
-        <img alt="" class="i-amphtml-story-attribution-logo" />
+const renderSystemLayerElement = (element, children, isHidden) => {
```
Comment (coreymasanto) [inline extensions/amp-story/1.0/amp-story-system-layer.js:98]: Note: despite the seemingly large diff, the only changes to this function are the following:
1. There is now an `isHidden` parameter
2. The system layer element is placed into a `systemLayerElement` variable before being returned
3. The 'i-amphtml-story-hidden' class is conditionally added at the very bottom of the function

File: extensions/amp-story/1.0/amp-story.js (hunk 1 above)
Comment (coreymasanto) [reply extensions/amp-story/1.0/amp-story.js:1706]: I've updated `amp-story-system-layer.js` to hide the system layer via the `i-amphtml-story-hidden` class, but it's not working as expected. The issue is that loading into `preview` mode correctly results in a hidden system layer, but transitioning to `visible` mode does not result in the system layer becoming shown.

Even though I can see that the preview -> visible transition results in `this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, true)` being correctly called within amp-story, `onSystemUiIsVisibleStateUpdate_` fails to be called within amp-story-system-layer

Comment (coreymasanto) [reply extensions/amp-story/1.0/amp-story.js:1706]: Fixed this with my latest commit

File: extensions/amp-story/1.0/amp-story-system-layer.js [hunk 4]
```diff
@@ -282,18 +292,27 @@ export class SystemLayer {
...
-      this.progressBar_.build(initialPageId)
+      this.progressBar_.build(initialPageId),
+      isHidden
     );
+    if (isHidden) {
+      // The default value of `SYSTEM_UI_IS_VISIBLE_STATE` is `true`. We set it
+      // to `false` here solely to ensure that a subsequent firing of the
+      // `TOGGLE_SYSTEM_UI_IS_VISIBLE` action with a `true` value registers as a
+      // change in state instead of being a no-op.
+      this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, false);
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-system-layer.js:314]: (Plz see other comment, this should be set somewhere else and the system layer component should only consume the SYSTEM_UI_IS_VISIBLE state. You're making assumptions that system-layer is the only SYSTEM_UI which is not necessarily true, it should be set by a higher level component and consumed by all others)

File: src/service/ampdoc-impl.js [hunk 5]
```diff
@@ -727,6 +727,14 @@ export class AmpDoc {
     return devAssert(this.visibilityState_);
   }
 
+  /**
+   * Whether the AMP document currently being previewed.
+   * @return {boolean}
+   */
+  isPreviewing() {
```
Comment (gmajoulet) [inline src/service/ampdoc-impl.js:734]: Nit: `isPreview()` to be consistent with the `PREVIEW` visibilityState

File: extensions/amp-story/1.0/amp-story-system-layer.js (hunk 4 above)
Comment (coreymasanto) [reply extensions/amp-story/1.0/amp-story-system-layer.js]: Makes sense, I've moved this dispatch to amp-story.js.

**Review by gmajoulet — APPROVED** [review]

**Comment by coreymasanto** [comment]
Adding @erwinmombay for `src/service/ampdoc-impl.js` OWNERS approval

**Review by erwinmombay — APPROVED** [review]

File: src/service/ampdoc-impl.js [hunk 6]
```diff
@@ -727,6 +727,14 @@ export class AmpDoc {
     return devAssert(this.visibilityState_);
   }
 
+  /**
+   * Whether the AMP document currently being previewed.
```
Comment (erwinmombay) [inline src/service/ampdoc-impl.js:731]: currently is being previewed?

Comment (coreymasanto) [reply src/service/ampdoc-impl.js:731]: I'll fix the comment in another PR, thanks for the review!

**Pull request merged**